pkg/store/bucket.go

```diff
@@ -377,6 +398,31 @@ func (s *BucketStore) numBlocks() int {
 	return len(s.blocks)
 }
 
+func (s *BucketStore) isBlockInMinMaxRange(ctx context.Context, id ulid.ULID) (bool, error) {
+	dir := filepath.Join(s.dir, id.String())
+
+	b := &bucketBlock{
```

Comment:
Allocating the whole block for quick check is not great. I think we should just extract `loadMeta` method to separate function and user.

Also at some point we need some meta syncer/ matcher that deals with all of this, something similar to https://github.com/thanos-io/thanos/pull/1394 ): 

Also even with time slicing looks like we will download all meta.json files to disk. Is that what we want? Essentially time slices will not include meta.json files on disk.

Reply:
Good question, IMO those meta.json files are small in size, so why not? It will speed up our next meta.json lookup.